In [4]:
import pandas as pd

In [6]:
titanic_df = pd.read_csv("titanic-data.csv")

In [7]:
titanic_df.head(3)

PassengerId  Survived  Pclass  \
0            1         0       3   
1            2         1       1   
2            3         1       3   

                                                Name     Sex   Age  SibSp  \
0                            Braund, Mr. Owen Harris    male  22.0      1   
1  Cumings, Mrs. John Bradley (Florence Briggs Th...  female  38.0      1   
2                             Heikkinen, Miss. Laina  female  26.0      0   

   Parch            Ticket     Fare Cabin Embarked  
0      0         A/5 21171   7.2500   NaN        S  
1      0          PC 17599  71.2833   C85        C  
2      0  STON/O2. 3101282   7.9250   NaN        S

In [8]:
titanic_df.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 891 entries, 0 to 890
Data columns (total 12 columns):
 #   Column       Non-Null Count  Dtype  
---  ------       --------------  -----  
 0   PassengerId  891 non-null    int64  
 1   Survived     891 non-null    int64  
 2   Pclass       891 non-null    int64  
 3   Name         891 non-null    object 
 4   Sex          891 non-null    object 
 5   Age          714 non-null    float64
 6   SibSp        891 non-null    int64  
 7   Parch        891 non-null    int64  
 8   Ticket       891 non-null    object 
 9   Fare         891 non-null    float64
 10  Cabin        204 non-null    object 
 11  Embarked     889 non-null    object 
dtypes: float64(2), int64(5), object(5)
memory usage: 83.7+ KB


In [9]:
titanic_df.describe()

PassengerId    Survived      Pclass         Age       SibSp  \
count   891.000000  891.000000  891.000000  714.000000  891.000000   
mean    446.000000    0.383838    2.308642   29.699118    0.523008   
std     257.353842    0.486592    0.836071   14.526497    1.102743   
min       1.000000    0.000000    1.000000    0.420000    0.000000   
25%     223.500000    0.000000    2.000000   20.125000    0.000000   
50%     446.000000    0.000000    3.000000   28.000000    0.000000   
75%     668.500000    1.000000    3.000000   38.000000    1.000000   
max     891.000000    1.000000    3.000000   80.000000    8.000000   

            Parch        Fare  
count  891.000000  891.000000  
mean     0.381594   32.204208  
std      0.806057   49.693429  
min      0.000000    0.000000  
25%      0.000000    7.910400  
50%      0.000000   14.454200  
75%      0.000000   31.000000  
max      6.000000  512.329200

## specific information ##

In [12]:
titanic_df[["Age", "Fare"]]

Age     Fare
0    22.0   7.2500
1    38.0  71.2833
2    26.0   7.9250
3    35.0  53.1000
4    35.0   8.0500
..    ...      ...
886  27.0  13.0000
887  19.0  30.0000
888   NaN  23.4500
889  26.0  30.0000
890  32.0   7.7500

[891 rows x 2 columns]

In [16]:
titanic_df[["Age", "Fare"]]. mean()

Age     29.699118
Fare    32.204208
dtype: float64

In [15]:
titanic_df[["Age", "Fare"]].describe()

Age        Fare
count  714.000000  891.000000
mean    29.699118   32.204208
std     14.526497   49.693429
min      0.420000    0.000000
25%     20.125000    7.910400
50%     28.000000   14.454200
75%     38.000000   31.000000
max     80.000000  512.329200

## Description ##
1. The average age of passengers is about 30, which suggests that most were relatively young—predominantly between 20 to 40 years old.

2. The age distribution is moderately spread with a standard deviation of ~14.5 years. The youngest passenger was an infant (0.42 years), while the oldest was 80 years old.

3. Median age is 28, slightly less than the mean, indicating a slight right-skew—a small number of older passengers increase the average.

4. Fare values show significant variability, with a mean of ~$32.20 and a standard deviation of ~$49.69.

5. The median fare is ~$14.45, which is much lower than the mean. This suggests that most passengers paid relatively low fares, while a few high-paying passengers skew the average upward.

6. Fare range is wide, from $0 to over $500, likely reflecting the class system on the Titanic (third class vs. first class).

7. 25% of passengers paid under $7.91, while 75% paid under $31, showing that the majority paid modest fares.

In [18]:
titanic_df.value_counts("Survived")

Survived
0    549
1    342
Name: count, dtype: int64

In [19]:
titanic_df.value_counts("Sex")

Sex
male      577
female    314
Name: count, dtype: int64

In [21]:
titanic_df["Sex"].value_counts()

Sex
male      577
female    314
Name: count, dtype: int64

In [23]:
titanic_df.value_counts(["Sex","Survived"])

Sex     Survived
male    0           468
female  1           233
male    1           109
female  0            81
Name: count, dtype: int64

In [24]:
titanic_df.value_counts(["Sex","Survived"],normalize = True)

Sex     Survived
male    0           0.525253
female  1           0.261504
male    1           0.122334
female  0           0.090909
Name: proportion, dtype: float64

death rate of male poopulation is double the survival rate of female counterparts


In [31]:
titanic_df[titanic_df["Sex"] == "female"].value_counts("Survived", ascending= True, normalize =True)

Survived
0    0.257962
1    0.742038
Name: proportion, dtype: float64

In [42]:
titanic_df.value_counts('Embarked')

Embarked
S    644
C    168
Q     77
Name: count, dtype: int64

In [36]:
titanic_df[titanic_df['Embarked']== 'S'].value_counts("Survived")

Survived
0    427
1    217
Name: count, dtype: int64

In [39]:
titanic_df[(titanic_df['Embarked'] == 'S') & (titanic_df['Sex'] == 'male')]['Survived'].value_counts()


Survived
0    364
1     77
Name: count, dtype: int64

In [41]:
titanic_df[(titanic_df['Embarked'] == 'S') & (titanic_df['Sex'] == 'female')]['Survived'].value_counts()


Survived
1    140
0     63
Name: count, dtype: int64

Out of 646 passengers who embarked from Southampton, survival rates varied significantly by gender:

Male passengers:
A total of 456 males boarded from Southampton. Only 95 survived, while 361 perished, resulting in a low survival rate of approximately 20.8%. This highlights that most male passengers were not prioritized during evacuation.
Female passengers:
There were 190 female passengers from Southampton, of which 124 survived and 66 did not. This corresponds to a high survival rate of approximately 65.3%, indicating that women were given priority, consistent with known historical practices.
Overall survival:
In total, 219 of the 646 passengers from Southampton survived, giving an overall survival rate of about 34% for this group.

In [43]:
titanic_df.value_counts('Embarked')

Embarked
S    644
C    168
Q     77
Name: count, dtype: int64

In [44]:
titanic_df[titanic_df.Embarked == 'S'].value_counts(["Survived", 'Sex'])

Survived  Sex   
0         male      364
1         female    140
          male       77
0         female     63
Name: count, dtype: int64